Source/Protocols/Request.swift

```diff
@@ -6,24 +6,32 @@ public typealias EmptyResponse = Void
 
 public protocol Request {
   typealias ResponseObject
+  typealias ResponseError = NSError
   func build() -> NSURLRequest
-  func parse(j: JSON) -> Result<ResponseObject, NSError>
+  func parse(j: JSON) -> Result<ResponseObject, SwishError>
+  func transformError(error: SwishError) -> ResponseError
```

Comment:
Is the intended use case for this to allow clients to map `SwishError` directly into their own error types? What if this function instead accepted a function `<E: ErrorType> SwishError -> E`?